src/makeCalculator.js

```diff
@@ -4,7 +4,43 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculate = {
+    result: 0,
+
+    add(number) {
+      this.result += number;
+    },
+
+    subtract(number) {
+      this.result -= number;
+    },
+
+    multiply(number) {
+      this.result *= number;
+    },
+
+    divide(number) {
+      this.result /= number;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    operate(callback, number) {
+      if (calculate.hasOwnProperty(callback) === false) {
```

Comment:
The condition in the `operate` method is incorrect. You are checking if the `calculate` object has a property with the name of the callback function, but you should be checking if the callback is a function. Replace `calculate.hasOwnProperty(callback)` with `typeof callback === 'function'` to ensure that the passed argument is indeed a function.